### PR TITLE
Error for woo 3+ and docs

### PR DESCRIPTION
**Pull request by bahiirwa**
Since Woocommerce mega major Update 3.0+ things have changed quite a lot:

WC_Order properties can't be access directly as before and will throw some errors.
New WC_Order and C_Abstract_Order getters and setters methods are now needed. There is some New classes for Order items: WC_Order_Item class.
WC_Order_Item_Product class.
So also the Order items properties will not be accessible as before in a foreach loop and you will have to use this specific getter and setter methods instead.

Using some WC_Order and WC_Abstract_Order methods (example):

// Get an instance of the WC_Order object (same as before)
$order = wc_get_order( $order_id );

### CODE REVIEW

**Comment by Jakeii** [comment]
Looks good, thanks for the contribution! But is it backwards compatible?

File: gateway.php [hunk 1]
```diff
@@ -328,7 +328,7 @@ public function admin_options() { ?>
         function thankyou_page($order_id) {
           global $woocommerce;
           
-          $order = new WC_Order( $order_id );
+          $order = new wc_get_order( $order_id );
```
Comment (akhenda) [inline gateway.php]: The new keyword is unnecessary here.

File: gateway.php [hunk 2]
```diff
@@ -480,7 +480,7 @@ function create_url($order_id){
          **/
         function pesapal_xml($order_id) {
           
-          $order                      = new WC_Order( $order_id );
+          $order                      = new wc_get_order( $order_id );
           $pesapal_args['total']      = $order->get_total();
           $pesapal_args['reference']  = $order_id;
           $pesapal_args['first_name'] = $order->billing_first_name;
```
Comment (akhenda) [inline gateway.php]: We should probably also change this to `$order->get_billing_first_name()`. In WooCommerce 3.0+, we should not access these properties directly.

**Review by akhenda — COMMENTED** [review]
Other than the following comments, the PR looks good to me.

**Comment by Jakeii** [comment]
Thanks guys, I'm going to merge this, no need to support <3.0.0 I suppose. But I'll increment a major version.

**Comment by bahiirwa** [comment]
Kindly update the plugin in the SVN repo. It 's says you havent updated in two years.

**Comment by Jakeii** [comment]
@bahiirwa I've done this today.

**Pull request merged**